include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -1029,6 +1029,14 @@ struct OrtApi {
   */
   ORT_API2_STATUS(SetLanguageProjection, _In_ const OrtEnv* ort_env, _In_ OrtLanguageProjection projection);
 
+  /**
+   * Add a pre-allocated initializer to a session. If a model contains an initializer with a name
+   * that is same as the name passed to this API call, ORT will use this initializer instance
+   * instead of deserializing one from the model file. This is useful when you want to share
+   * the same initializer across sessions.
+   */
+  ORT_API2_STATUS(AddInitializer, _Inout_ OrtSessionOptions* options, _In_z_ const char* name,
+                  _In_ const OrtValue* val);
 };
```

Comment:
Do we take ownership of the OrtValue? If so doco should call that out and say it needs to be non-const so we can free it?